### PR TITLE
examples: Add an example for scale functionality.

### PR DESCRIPTION
**Pull request by bratish**
Added a simple example for  scale functionality.

Tested on Linux (Pop!_OS 21.04).

### CODE REVIEW

File: examples/scale/main.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
+use gtk::prelude::*;
```
Comment (bilelmoussaoui) [inline examples/scale/main.rs:1]: Please run cargo fmt on your code.

File: examples/scale/main.rs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            false, 
+            move |args| {
+            let slider = args[0]
+                .get::<Scale>()
+                .expect("The value needs to be of type `Scale`.");
+
+            println!("Horizontal scale value: {:?}", slider.value());
+            None
+        })
+        .expect("Could not connect to signal.");
```
Comment (bilelmoussaoui) [inline examples/scale/main.rs:67]:
```suggestion
    horizontal_scale. connect_value_changed(move |slider {
        println!("Horizontal scale value: {:?}", slider.value());
    });
```
why not just?

Comment (bilelmoussaoui) [reply examples/scale/main.rs:67]: Maybe display those values on a label?

File: examples/scale/main.rs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            false, 
+            move |args| {
+            let slider = args[0]
+                .get::<Scale>()
+                .expect("The value needs to be of type `Scale`.");
+
+            println!("Vertical scale value: {:?}", slider.value());
+            None
+        })
+        .expect("Could not connect to signal.");
```
Comment (bilelmoussaoui) [inline examples/scale/main.rs:88]:
```suggestion
    vertical_scale.connect_local(
            "value-changed", 
            false, 
            move |args| {
            let slider = args[0]
                .get::<Scale>()
                .expect("The value needs to be of type `Scale`.");

            println!("Vertical scale value: {:?}", slider.value());
            None
        })
        .expect("Could not connect to signal.");
```
Same as the horizontal scale

File: examples/scale/main.rs [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            ApplicationWindow, 
+            Grid, 
+            Label, 
+            Orientation, 
+            Scale
+        };
+
+fn main() {
+    let application = Application::new(
+        Some("com.github.gtk-rs.examples.builder_pattern"),
```
Comment (bilelmoussaoui) [inline examples/scale/main.rs:13]:
```suggestion
        Some("com.github.gtk-rs.examples.scale"),
```

File: examples/scale/main.rs [hunk 5]
```diff
@@ -0,0 +1,108 @@
+use gtk::prelude::*;
+use gtk::{  Adjustment, 
+            Application, 
+            ApplicationWindow, 
+            Grid, 
+            Label, 
+            Orientation, 
+            Scale
+        };
```
Comment (bilelmoussaoui) [inline examples/scale/main.rs:9]: I definitely don't like importing everything here. In general, I would do that unless the type/function/macro is used more than once.

File: examples/scale/README.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+# Basics
+
+This example demonstrates how to create `scales` and place them in the `window` using a grid. It also shows how to implement `callbacks`.
```
Comment (bilelmoussaoui) [inline examples/scale/README.md:3]:
```suggestion
This example demonstrates how to create `gtk::Scale` and place them in the `gtk::Window` using a grid.
```
Something like this?

File: examples/scale/README.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+# Basics
```
Comment (bilelmoussaoui) [inline examples/scale/README.md:1]:
```suggestion
# Scale
```
maybe?
This also needs to be added to the main readme inside the examples directory

**Comment by bilelmoussaoui** [comment]
Thanks for the example! overall looks good to me. Do you mind taking the screenshot using the default GTK theme like the other examples?

**Comment by bratish** [comment]
Thank you for taking time and review the PR. I have tried to implement all the suggestions except the screenshot. The only computer I have access to is a Pop Os! and I could not change it to the theme to match the ones in the other examples (the default GTK theme). :(

Please let me know if this is okay. Thank you!

**Comment by bilelmoussaoui** [comment]
> Thank you for taking time and review the PR. I have tried to implement all the suggestions except the screenshot. The only computer I have access to is a Pop Os! and I could not change it to the theme to match the ones in the other examples (the default GTK theme). :(
> 
> Please let me know if this is okay. Thank you!

You can change the theme in the settings i believe and switch back once you took the screenshot. Sorry for the long delay, i forgot you actually replied.

**Comment by bilelmoussaoui** [comment]
Rebased and fixed on https://github.com/gtk-rs/gtk4-rs/pull/1569, thanks

**Pull request closed without merging**